new resources: azurerm_datadog_monitor_tag_rule azurerm_datadog_monitor_sso_configuration (#17825)
hashicorp/terraform-provider-azurerm: website/docs/r/datadog_tagrules.html.markdown

---
subcategory: "Datadog"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_datadog_monitor_tagrules"
description: |-

>   Manages TagRules on the datadog Monitor.

Reviewer: ```suggestion
  Manages Tag Rules on the datadog Monitor.
```

Author: Updated

---

> # azurerm_datadog_monitor_tagrules

Reviewer: Can we rename this to be `azurerm_datadog_monitor_tag_rule`?

Author: Since tagrules is a single input file, will the split be neccassary?

Reviewer: @vikotha At least the last `s` can be removed as we try to make the resource name as a singular.

Author: Updated

Manages TagRules on the datadog Monitor.

## Example Usage

### Adding TagRules on monitor
```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-datadog"
  location = "West US 2"
}
resource "azurerm_datadog_monitor_tagrules" "example" {
  name                = "example-monitor"
  resource_group_name = azurerm_resource_group.example.name
  log_rules{
    send_subscription_logs = true
  }
  metric_rules{
    filtering_tag {
        name = "Test"
        value = "Logs"
        action = "Include"
    }
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this datadog Monitor.

> * `resource_group_name` - (Required) The name of the Resource Group where the datadog Monitor should exist.

Reviewer: Can we use `datadog_monitor_id` instead?

Author: Done

---

> * `rule_set_name` - (Optional) The name of the Tag Rules configuration.

Reviewer: Can we rename this to `name`?

Author: Done

> * `log_rules` - (Optional) A `log_rules` block as defined below.

Reviewer: Can we rename this to `log`?

Author: It's a rule block, so removing it will create any confusion?

Reviewer: This resource itself is already a rule resource, so I assume the duplication on the "rule" here is less meaningful.

Author: Done

> * `metric_rules` - (Optional) A `metric_rules` block as defined below.

Reviewer: Can we rename this to `metric`?

Author: Updated

---

An `log_rules` block supports the following:

> * `send_aad_logs` - (Optional) Boolean flag specifying if AAD logs should be sent for the Monitor resource.

Reviewer: Can we rename this to `aad_log_enabled` for consistency across the provider?

Author: Updated

> * `send_subscription_logs` - (Optional) Boolean flag specifying if Azure subscription logs should be sent for the Monitor resource.

Reviewer: Can we rename this to `subscription_log_enabled` for consistency across the provider?

Author: Updated

> * `send_resource_logs` - (Optional) Boolean flag specifying if Azure resource logs should be sent for the Monitor resource.

Reviewer: Can we rename this to `resource_log_enabled` for consistency across the provider?

Author: Updated

> **NOTE:** List of filtering tags to be used for capturing logs. This only takes effect if SendResourceLogs flag is enabled. If empty, all resources will be captured. If only Exclude action is specified, the rules will apply to the list of all available resources. If Include actions are specified, the rules will only include resources with the associated tags.
---

A `metric_rules` block supports the following:

* `filtering_tag` - (Optional) A `filtering_tag` block as defined below.

> **NOTE:** List of filtering tags to be used for capturing metrics. If empty, all resources will be captured. If only Exclude action is specified, the rules will apply to the list of all available resources. If Include actions are specified, the rules will only include resources with the associated tags.
---

A `filtering_tag` block supports the following:

* `name` - (Required) Name of the Tag.

* `value` - (Required) Value of the Tag.

* `action` - (Required) Allowed values Enable or Disbale.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported:

* `id` - The ID of the Tagrules on datadog monitor.

> * `type` - The type of the monitor resource.

Reviewer: This can be removed.

Author: Done

> * `provisioning_state` - The state of Datadog monitor.

Reviewer: This can be removed.

Author: done

---
subcategory: "Datadog"
layout: "azurerm"

> page_title: "Azure Resource Manager: azurerm_datadog_monitor_tagrule"

Reviewer: I still prefer renaming this to `azurerm_datadog_monitor_tag_rule`.

Author: Updated

description: |-

>   Manages Tag Rules on the datadog Monitor.

Reviewer: ```suggestion
  Manages Tag Rules on the Datadog Monitor.
```

Author: Done

---

# azurerm_datadog_monitor_tagrule

Manages TagRules on the datadog Monitor.

## Example Usage

### Adding TagRules on monitor
```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-datadog"
  location = "West US 2"
}
resource "azurerm_datadog_monitor_tagrule" "example" {
  datadog_monitor_id = azurerm_datadog_monitor.example.id
  log {
    subscription_log_enabled = true
  }
  metric {
    filtering_tag {
      name   = "Test"
      value  = "Logs"
      action = "Include"
    }
  }
}
```

## Arguments Reference

The following arguments are supported:

> * `datadog_monitor_id` - (Required) The Datadog Monitor Id which should be used for this.

Reviewer: ```suggestion
* `datadog_monitor_id` - (Required) The Datadog Monitor Id which should be used for this Datadog Monitor Tag Rule. Changing this forces a new Datadog Monitor Tag Rule to be created.
```

Author: Added

> * `resource_group_name` - (Required) The name of the Resource Group where the datadog Monitor should exist.

Reviewer: This can be removed?

Author: Removed

---

* `rule_set_name` - (Optional) The name of the Tag Rules configuration.

> * `log` - (Optional) A `log_rules` block as defined below.

Reviewer: ```suggestion
* `log` - (Optional) A `log` block as defined below.
```

Author: Done

> * `metric` - (Optional) A `metric_rules` block as defined below.

Reviewer: ```suggestion
* `metric` - (Optional) A `metric` block as defined below.
```

Author: Done

---

An `log` block supports the following:

> * `aad_log_enabled` - (Optional) Boolean flag specifying if AAD logs should be sent for the Monitor resource.

Reviewer: Can we refine the tone here to make it consistent with the other document?

```suggestion
* `aad_log_enabled` - (Optional) Whether AAD logs should be sent for the Monitor resource?
```

Author: Changed

> * `subscription_log_enabled` - (Optional) Boolean flag specifying if Azure subscription logs should be sent for the Monitor resource.

Reviewer: ```suggestion
* `subscription_log_enabled` - (Optional) Whether Azure subscription logs should be sent for the Monitor resource?
```

Author: Changed

> * `resource_log_enabled` - (Optional) Boolean flag specifying if Azure resource logs should be sent for the Monitor resource.

Reviewer: ```suggestion
* `resource_log_enabled` - (Optional) Whether Azure resource logs should be sent for the Monitor resource?
```

Author: Changed

> * `filtering_tag` - (Optional) A `filtering_tag` block as defined below.

Reviewer: Can we just rename this to `filter` given the current resource is a tag rule?

Author: Done

> > **NOTE:** List of filtering tags to be used for capturing logs. This only takes effect if SendResourceLogs flag is enabled. If empty, all resources will be captured. If only Exclude action is specified, the rules will apply to the list of all available resources. If Include actions are specified, the rules will only include resources with the associated tags.

Reviewer: I'm a bit consfused about the meaning of this note. What does `SendResourceLogs` flag, `Exclude`/Include` action mean here?

Author: Updated the SendResourceLogs to "resource_log_enabled".
Exclude/Include action is based on the tags attached the resources in Azure.

---

A `metric` block supports the following:

> * `filtering_tag` - (Optional) A `filtering_tag` block as defined below.

Reviewer: Can we just rename this to `filter` given the current resource is a tag rule?

Author: Done

> **NOTE:** List of filtering tags to be used for capturing metrics. If empty, all resources will be captured. If only Exclude action is specified, the rules will apply to the list of all available resources. If Include actions are specified, the rules will only include resources with the associated tags.
---

A `filtering_tag` block supports the following:

* `name` - (Required) Name of the Tag.

* `value` - (Required) Value of the Tag.

> * `action` - (Required) Allowed values Enable or Disable.

Reviewer: Given this is a boolean value, can we just change it to `enabled` with type bool?

Author: Typo in description. Changed it to "Include" and "Exclude"

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:

* `create` - (Defaults to 30 minutes) Used when creating the Tagrules on the datadog Monitor.
* `read` - (Defaults to 5 minutes) Used when retrieving the Tagrules on the datadog Monitor.
* `update` - (Defaults to 30 minutes) Used when updating the Tagrules on the datadog Monitor.

> * `delete` - (Defaults to 30 minutes) Used when deleting the Tagrules on the datadog Monitor.

Reviewer: Can this resource be imported? If yes, please also add the "Import" section?

Author: Added

Author: Done
